XbSymbolDatabase.c

```diff
@@ -187,12 +187,27 @@ void XbSymbolBypassRevLimit(bool bypass_limit)
     bStrictRevisionLimit = !bypass_limit;
 }
 
-void XbSymbolOutputMessage(xb_output_message mFlag, const char* message) {
+void XbSymbolOutputMessage(xb_output_message mFlag, const char* message)
+{
     if (output_func != 0) {
         output_func(mFlag, message);
     }
 }
 
+void XbSymbolOutputMessageFormat(xb_output_message mFlag, const char* format, ...)
```

Comment:
I don't think this needs to be a separate function, we can just extend `XbSymbolOutputMessage`. The old style of calling it is still valid, and it's not exactly performance critical either.

Reply:
> The old style of calling it is still valid

Format string as message perform things differently plus bad code can cause an exception without awareness. Plus C standard doesn't allow two same function names with different parameters.

> it's not exactly performance critical either.

Fee free to scatter many `tempBuffer[2048]`and many of snprintf/printf calls around in multiple functions. I'm sure the stacks will love multiple big local variable sizes. 🙄

Reply:
> Format string as message perform things differently plus bad code can cause an exception without awareness.

C doesn't have exceptions.

*(The most dangerous part in your implementation would be a buffer-overflow, which can happen regardless)*

> Plus C standard doesn't allow two same function names with different parameters.

I did not mean an overload: You can change the original function (as you also implied in the second half of your response).

Unless we did `XbSymbolOutputMessage(buffer)` which might allow "%" to be present, this is a non-issue.

This is similar to how you'd always use `printf("Hello World!\n")` instead of `puts("Hello World!\n")`. It's not an issue unless you write dangerous code to begin with.

This is what I meant with "The old style of calling it is still valid".

> Fee free to scatter many tempBuffer[2048]and many of snprintf/printf calls around in multiple functions. I'm sure the stacks will love multiple big local variable sizes.

This function is not recursive. By the time it leaves, the memory is already gone again.

As this is the bottom of the stack, and not initialized, this reduces to a single add/sub instruction - it doesn't cost performance except for a page fault (if you are unlucky), which doesn't matter either because it's not a hot-loop and will itself call all over the place (`output_func`).

> EDIT: Also, we don't need to waste extra CPU cycles for formatting the string if it's not going to be output at all.

I don't think a bit of performance output critical, especially as these functions are only ran once and the result is even cached in most applications.

I'd also assume that compilers optimize the `vsprintf` call. All popular compilers definitely *do* optimize `printf` into `puts` if no `%` is present at compile time; I assume the same happens to `vsprintf` which probably turns into `strcpy` (I did not check though).